numpy/random/src/distributions/distributions.c

```diff
@@ -960,7 +960,15 @@ RAND_INT_TYPE random_geometric_search(bitgen_t *bitgen_state, double p) {
 }
 
 int64_t random_geometric_inversion(bitgen_t *bitgen_state, double p) {
-  return (int64_t)ceil(-random_standard_exponential(bitgen_state) / npy_log1p(-p));
+  double z = ceil(-random_standard_exponential(bitgen_state) / npy_log1p(-p));
+  /*
+   * The constant 9.223372036854776e+18 is the smallest double that is
+   * larger than INT64_MAX.
+   */
+  if (z >= 9.223372036854776e+18) {
```

Comment:
Why not `z > (double)INT64_MAX`?

Reply:
`(double)INT64_MAX` and `9.223372036854776e+18` are the same value, so yeah, that spelling also works.  The comparison must be `>=`, though, because the value of `(double)INT64_MAX` is actually `INT64_MAX + 1` (i.e. `2**63`) and we don't want to allow that value to get through and invoke undefined behavior.

Reply:
Ah, right, because of the spacing. The literal is probably clearer than the comment required for `z >= (double)INT64_MAX` would be.

Reply:
By "the literal", do you mean `9.223372036854776e+18`?

Reply:
Yes.